engine/src/main/java/org/terasology/persistence/internal/EntitySetDeltaRecorder.java

```diff
@@ -83,9 +92,12 @@ private EntityDelta getOrCreateEntityDeltaFor(EntityRef entity) {
         return entityDelta;
     }
 
-    public void onEntityDestroyed(long entityId) {
-        entityDeltas.remove(entityId);
-        destroyedEntities.add(entityId);
+    public void onEntityDestroyed(EntityRef entity) {
+        if (!entity.isPersistent()) {
```

Comment:
There are quite some discussions about "guard statement" vs "singe point of return" paradigm:
e.g.
http://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement

I personally have seen large methods, in which multiple return statements spread out through the method have caused the introduction of bugs. (e.g. a return statement got added later on at the wrong place). The main issue there was however not the usage of multiple return statements but the complexity of the method. 

Gurad statements can help to increase the readability of code:
* The nice thing about guard statements is that you can read them and almost immediately forget them when you read the next code.
* If you use an if (condition) { rest of method } then the reader has to remember the condition till the rest of the method: It could be that there is an else block
* Especially if multiple conditions are to check, using pure "single point of return" can result in some overly complex methods.

In the particular case of code the usage of the guard statement also highlights the fact that multiple methods use the same check.